torch/_torch_docs.py

```diff
@@ -3152,7 +3152,24 @@ def merge_dicts(*dicts):
            r"""
 flatten(input, start_dim=0, end_dim=-1) -> Tensor
 
-Flattens a contiguous range of dims in a tensor.
+By default, flattens :attr:`input` into a one-dimensional tensor with the same number of elements.
+If :attr:`start_dim` or :attr:`end_dim` are passed, only dimensions included in the range are
+flattened.
+
+Flattening an :attr:`input` with shape :math:`(l*, f+, r*)` produces an output shape of
+:math:`(l*, \prod f+, r*)` where :math:`l*` is the sizes of the zero or more "left" dimensions,
+:math:`f+` is the sizes of the one or more dimensions between :attr:`start_dim` and :attr:`end_dim` that
+will be "flattened", and :math:`r*` is the sizes of the zero or more "right" dimensions in :attr:`input`.
+
+Like :func:`torch.reshape`, flatten returns a view when possible.
+In fact, this operation is equivalent to calling :math:`\text{torch.reshape}(\textit{input}, (l*, \prod f+, r*))`
+unless the computed output shape is equivalent to the input shape, i.e., `start_dim == end_dim`.
+
+Unlike :func:`torch.reshape` which always returns a new tensor object, if `start_dim == end_dim`
+this function returns the original object :attr:`input`.
+
+.. note::
+    Flattening a zero-dimensional tensor will always return a one-dimensional view.
```

Comment:
."will always return a one-dimensional view." -> "returns a one-dimensional view of :attr:\`input\`"

Reply:
That's not necessarily true. The returned tensor's base is not necessarily input. They could both view a common base.

Reply:
This is a great technical point that our documentation is probably very lazy about.